ansible/roles/opencontrail_provision/tasks/master.yml

```diff
@@ -0,0 +1,82 @@
+---
+- name: Ensure API server is running
+  wait_for: port=8082
+
+- name: Create temporary directory
+  command: mktemp -d
+  register: mktemp_result
+  always_run: yes
+
+- name: Temporary directory var
+  set_fact:
+    host_tmp_dir: "{{ mktemp_result.stdout }}"
+  always_run: yes
+
+- name: Namespace file
+  copy: src=namespace.yml dest={{ host_tmp_dir }}
+
+- name: Namespace status
+  command: kubectl get namespace opencontrail
+  ignore_errors: yes
+  register: namespace
+
+- name: Create namespace
+  command: kubectl create -f "{{ host_tmp_dir }}/namespace.yml"
+  when: namespace is defined and namespace.rc != 0
+
+- name: Cleanup
+  file: path="{{ host_tmp_dir }}" state=absent
+  always_run: yes
+
+- set_fact:
+    opencontrail_config_http_proxy_host: "{{ opencontrail_http_proxy | regex_replace('[a-z]+://([a-zA-Z0-9][a-zA-Z0-9\\.\\-_]{0,62})(:[0-9]+)?', '\\\\1') }}"
+  when: opencontrail_http_proxy is defined
+
+- set_fact:
+    opencontrail_config_http_proxy_address: "{{ opencontrail_config_http_proxy_host | ipaddr('address') }}"
+  when: "opencontrail_config_http_proxy_host is defined"
+
+# lookup module dig requires python-dns package
+- set_fact:
+    opencontrail_config_http_proxy_address: "{{ lookup('dig', opencontrail_config_http_proxy_host) }}"
+  when: "opencontrail_config_http_proxy_host is defined and not opencontrail_config_http_proxy_address is defined"
+
+- name: Metadata services
+  command: >
+    docker run opencontrail/config:2.20 /usr/share/contrail-utils/provision_linklocal.py
```

Comment:
@galthaus yes: opencontrail release should be a ansible var and it should be consistently applied across templates, etc... the README mentions it; but it is not implemented as of now. All the manifest files as well as upstart / systemd  files are affected since they specify the docker container tag.